.github/workflows/nightly.yml

```diff
@@ -26,82 +28,35 @@ jobs:
             build_type: Debug
             static: true
             container: '{ "image": "rippleci/clio_ci:latest" }'
-    uses: ./.github/workflows/build_impl.yml
-    with:
-      runs_on: ${{ matrix.os }}
-      container: ${{ matrix.container }}
-      conan_profile: gcc
-      build_type: ${{ matrix.build_type }}
-      code_coverage: false
-      static: ${{ matrix.static }}
-      unit_tests: true
-      integration_tests: true
-      clio_server: true
-      disable_cache: true
-
-  run_tests:
-    needs: build
-    strategy:
-      fail-fast: false
-      matrix:
-        include:
-          - os: macos15
-            conan_profile: apple_clang_16
-            build_type: Release
-            integration_tests: false
-          - os: heavy
-            conan_profile: gcc
-            build_type: Release
-            container:
-              image: rippleci/clio_ci:latest
-            integration_tests: true
-          - os: heavy
-            conan_profile: gcc
-            build_type: Debug
-            container:
-              image: rippleci/clio_ci:latest
-            integration_tests: true
-    runs-on: [self-hosted, "${{ matrix.os }}"]
-    container: ${{ matrix.container }}
-
-    services:
-      scylladb:
-        image: ${{ (matrix.integration_tests) && 'scylladb/scylla' || '' }}
-        options: >-
-          --health-cmd "cqlsh -e 'describe cluster'"
-          --health-interval 10s
-          --health-timeout 5s
-          --health-retries 5
 
     steps:
-      - name: Clean workdir
-        if: ${{ runner.os == 'macOS' }}
-        uses: kuznetsss/workspace-cleanup@80b9863b45562c148927c3d53621ef354e5ae7ce #v1.0
-
-      - uses: actions/download-artifact@v4
+      - name: Build
+        uses: ./.github/workflows/build_impl.yml
         with:
-          name: clio_tests_${{ runner.os }}_${{ matrix.build_type }}_${{ matrix.conan_profile }}
-
-      - name: Run clio_tests
-        run: |
-          chmod +x ./clio_tests
-          ./clio_tests
-
-      - uses: actions/download-artifact@v4
+          runs_on: ${{ matrix.os }}
+          container: ${{ matrix.container }}
+          conan_profile: gcc
+          build_type: ${{ matrix.build_type }}
+          code_coverage: false
+          static: ${{ matrix.static }}
+          unit_tests: true
+          integration_tests: true
+          clio_server: true
+          disable_cache: true
+
+      - name: Test
+        uses: ./.github/workflows/test_impl.yml
         with:
-          name: clio_integration_tests_${{ runner.os }}_${{ matrix.build_type }}_${{ matrix.conan_profile }}
```

Comment:
Integration tests were waiting for unit tests, but there is no need for that.
In a new design it doesn't happen because these are separate jobs